### PR TITLE
Update packets up to 2018-12-12

### LINKED ISSUE #2334: Assert error while creating character
**Describe the bug**

When creating a character assert error occurs on the char server

```[Error]: Reserved too small packet buffer for packet 0x006D with size 152, but must be size 157
[Error]: --- failed assertion --------------------------------------------
[Error]: socket.c:2138: '0' in function `socket_validateWfifo'
[Error]: ./char-server() [0x447788]
[Error]: ./char-server() [0x45b8ee]
[Error]: ./char-server() [0x40ace0]
[Error]: ./char-server() [0x40eb01]
[Error]: ./char-server() [0x45afcb]
[Error]: ./char-server(main+0x2ba) [0x4073ba]
[Error]: /lib64/libc.so.6(__libc_start_main+0xea) [0x7f46cc60ef6a]
[Error]: ./char-server(_start+0x2a) [0x40748a]
[Error]: --- end failed assertion ----------------------------------------
```

**To Reproduce**
1. Create character
2. look at the char server

**Screenshots**
none

**System specs (please complete the following information):**
 - OS: [windows 10 ]
 - Hercules Version v2018.11.18+1 
 - Mode: [renewal]
 - Packet version: [20181017]
 - Client type: [main]

**Plugins used or source modifications**
none

**Additional context**
none

### PR DESCRIPTION
**Pull request by 4144**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Add packet CA_ACK_MOBILE_OTP.
Add packet AC_REQ_MOBILE_OTP.
Add packet CA_OTP_CODE.
Add packet ZC_ENCHANT_EQUIPMENT.
Add script command enchantitem.
Add packet ZC_SERVICE_MESSAGE_COLOR.
Add script command servicemessage.
Fix packet ZC_PROPERTY_HOMUN.
Fix constant MAX_CHAR_BUF for all packet versions (fixes #2334)
Update packet ZC_AUTOSPELLLIST for main client.
Add packets CZ_START_USE_SKILL/CZ_STOP_USE_SKILL.
Update packet ZC_NPC_MARKET_OPEN.
Update packet CZ_PARTY_CONFIG and fix weight field.
Add packet CZ_CAMERA_INFO (command /viewpointvalue and /setcamera)
Add validation for function char_mmo_char_tobuf.
Add support for int32 (4 bytes) item id sizes in zero clients and main clients.
Add packets, script commands and scripts for inventory expansion.
Implement inventory expansion feature. http://imgc1.gnjoy.com/ufile/ro/2018/11/21/QK3O60RA1MGOMU301NUO.gif (partially can works in any clients)
Move packets structs in login server into separate files.
Update packets up to 2018-12-12.
Update messages up to 2018-12-12.
Update keys up to 2018-12-12.

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2324)

**Comment by HerculesWSAPI** [comment]
This pull request **introduces 1 alert** when merging b09886dc2dafcfadcfae205c4901cb5a1e7ba908 into 97606eaf71983a82463062b04c2ec1cf5b078d57 - [view on LGTM.com](https://lgtm.com/projects/g/HerculesWS/Hercules/rev/pr-0a624d8bb6f4af190a4954b7e95714aad8f50ae0)

**new alerts:**

* 1 for Missing return statement

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/map/clif.c [hunk 1]
```diff
@@ -21992,17 +21992,20 @@ static void clif_item_preview(struct map_session_data *sd, int n)
...
 
 // insert cardId into equipped item in pos equipment slot into slot cardSlot.
-static void clif_enchant_equipment(struct map_session_data *sd, enum equip_pos pos, int cardSlot, int cardId)
+static bool clif_enchant_equipment(struct map_session_data *sd, enum equip_pos pos, int cardSlot, int cardId)
 {
 #if PACKETVER_MAIN_NUM >= 20160831 || PACKETVER_RE_NUM >= 20151118 || defined(PACKETVER_ZERO)
-	nullpo_retv(sd);
-	Assert_retv(cardSlot >= 0 && cardSlot <= 3);
+	nullpo_ret(sd);
+	Assert_ret(cardSlot >= 0 && cardSlot <= 3);
```
Comment (Asheraf) [inline src/map/clif.c]: using MAX_SLOTS would be better

File: src/map/script.c [hunk 2]
```diff
@@ -24790,6 +24790,56 @@ static BUILDIN(itempreview)
...
+	const int n = sd->equip_index[pos];
+	if (n < 0) {
+		ShowError("Item in equipment slot %d is not equipped\n", pos);
+		script->reportfunc(st);
+		script->reportsrc(st);
+		script_pushint(st, false);
+		return true;
+	}
+	const int cardSlot = script_getnum(st, 3);
+	if (cardSlot < 0 || cardSlot > 3) {
```
Comment (Asheraf) [inline src/map/script.c]: MAX_SLOTS here as well

**Pull request merged**